routes.js

```diff
@@ -11,27 +12,62 @@ const responseSchema = {
       data: {
         type: 'object',
         additionalProperties: true
+      },
+      errors: {
+        type: 'object',
+        properties: {
+          message: { type: 'string' },
+          locations: {
+            type: 'array',
+            items: {
+              type: 'object',
+              properties: {
+                line: { type: 'integer' },
+                column: { type: 'integer' }
+              }
+            }
+          },
+          path: {
+            type: 'array',
+            items: { type: 'string' }
+          }
+        }
       }
-      // TODO make this include errors
     }
   }
 }
 
-module.exports = async function (app, opts) {
-  app.setErrorHandler(async function (err, request, reply) {
-    if (!err.statusCode) {
-      throw err
-    }
+async function defaultErrorHandler (err, request, reply) {
+  if (!err.statusCode) {
+    throw err
+  }
 
-    reply.code(err.statusCode)
-    if (err.errors) {
-      return { errors: err.errors }
-    } else {
-      return {
-        message: err.message
-      }
+  reply.code(err.statusCode)
+  if (err.errors) {
+    const errors = err.errors.map(error => {
+      return error instanceof GraphQLError ? formatError(error) : { message: error.message }
+    })
+
+    return { errors }
+  } else {
+    return {
+      errors: [
+        { message: err.message }
+      ]
     }
-  })
+  }
+}
+
+function validationHandler (validationError) {
+  if (validationError) {
+    const err = new BadRequest()
+    err.errors = [validationError]
+    throw err
+  }
+}
+
+module.exports = async function (app, opts) {
+  app.setErrorHandler(opts.errorHandler || defaultErrorHandler)
```

Comment:
```suggestion
  if (typeof opts.errorHandler === 'function') {
    app.setErrorHandler(opts.errorHandler)
  } else if (opts.errorHandler === true || opts.errorHandler === undefined) {
      app.setErrorHandler(defaultErrorHandler)
  }
```

This is the behavior I'm expecting. A test passing `errorHandler: false`  and `errorHandler: true` must be added.